### PR TITLE
db: don't ignore range key bounds during excise

### PR DESCRIPTION
**Pull request by bananabrick**

### CODE REVIEW

File: testdata/excise [hunk 1]
```diff
@@ -162,3 +162,39 @@ would excise 2 files, use ingest-and-excise to excise.
...
+
+build ext2
+set z z
+----
+
+ingest-and-excise ext2 excise=b-c
+----
+
+# TODO(bananabrick): This looks insanely incorrect. Why is the second sstable
+# bound d-d?
```
Comment (itsbilal) [inline testdata/excise:193]: TODO should be taken out now?

**Review by itsbilal — APPROVED** [review]
Thanks for finding and fixing this quickly! LGTM :shipit:

File: testdata/excise (hunk 1 above)
Comment (bananabrick) [reply testdata/excise:193]: Lol my bad

**Comment by bananabrick** [comment]
race build probably not going to fail due to this. merging.

**Pull request merged**